### ISSUE TITLE
List customers with query filter causes fatal error

### ISSUE THREAD

**Issue by matt-allan** [helpscout/helpscout-api-php#175]
Hello,

I have been using version 2.3.3 successfully since 2019-06-07 but just today started receiving fatal errors. I am able to reproduce this issue on the master branch as well.

Since none of the code changed it looks like a change was made to the API that is triggering this issue, but I didn't see anything in the [changelog](https://developer.helpscout.com/mailbox-api/changelog/) that seemed related.

To reproduce the issue you can run the following script from the CLI:

```php
<?php

use HelpScout\Api\ApiClientFactory;
use HelpScout\Api\Customers\CustomerFilters;

require __DIR__ . '/vendor/autoload.php';

$client = ApiClientFactory::createClient();

$client->useClientCredentials(getenv('HELPSCOUT_APP_ID'), getenv('HELPSCOUT_APP_SECRET'));

$filter = (new CustomerFilters())
	->withQuery('(email:"matt.allan@somedomain.com")');

$customers = $client->customers()->list($filter);

var_dump($customers);
```

```
HELPSCOUT_APP_ID={your-app-id} HELPSCOUT_APP_SECRET={your-app-secret} php ./issue.php
PHP Fatal error:  Uncaught TypeError: Return value of HelpScout\Api\Http\Hal\HalDocument::getEmbedded() must be of the type array, object returned in /home/matt/code/helpscout-api-php/src/Http/Hal/HalDocument.php:54
Stack trace:
#0 /home/matt/code/helpscout-api-php/src/Http/Hal/HalDeserializer.php(62): HelpScout\Api\Http\Hal\HalDocument->getEmbedded('customers')
#1 /home/matt/code/helpscout-api-php/src/Http/RestClient.php(159): HelpScout\Api\Http\Hal\HalDeserializer::deserializeResources('HelpScout\\Api\\C...', 'customers', Object(HelpScout\Api\Http\Hal\HalDocument))
#2 /home/matt/code/helpscout-api-php/src/Customers/CustomersEndpoint.php(93): HelpScout\Api\Http\RestClient->getResources('HelpScout\\Api\\C...', 'customers', '/v2/customers?q...')
#3 /home/matt/code/helpscout-api-php/src/Customers/CustomersEndpoint.php(77): HelpScout\Api\Customers\CustomersEndpoint->loadCustomers('/v2/customers?q...', Object(HelpScout\Api\Customers\CustomerRequest))
#4 /home/matt/code/helpscout-api-php/issue.php(15): HelpScout\Api\Customers\Custo in /home/matt/code/helpscout-api-php/src/Http/Hal/HalDocument.php on line 54
```

**Comment by bob983** [comment]
@matt-allan Hi Matt, I'm the API engineer here in Help Scout but sadly I'm not fluent in our PHP SDK. There was a small update to the API today - it was supposed to be invisible - before the change when a List Customers filter returned no results the response did not include `_embedded.customers` array - now it contains empty array: `{"_embedded" : { "customers" : [] }}`.

Even when I use a filter that corresponds to just one result, I can still see the same structure: `{"_embedded" : { "customers" : [{....}] }}`. How does it look on your end? Could you dump the raw API response on your end?

There's clearly something going on but I'm not sure what it is yet.

Thanks!

**Comment by tucksaun** [comment]
@bob983 we are also experiencing this issue. When the filter matches an existing user, we don't experience any trouble.

Here is a dump of the response causing the issue for us:
```
{"_embedded":{"customers":[]},"_links":{"first":{"href":"https://api.helpscout.net/v2/customers?query=email:\"anonymous@anonymous.com\"&page=1"},"last":{"href":"https://api.helpscout.net/v2/customers?query=email:\"anonymous@anonymous.com\"&page=1"},"page":{"href":"https://api.helpscout.net/v2/customers?query=email:\"anonymous@anonymous.com\""},"self":{"href":"https://api.helpscout.net/v2/customers?query=email:\"anonymous@anonymous.com\""}},"page":{"size":50,"totalElements":0,"totalPages":0,"number":1}}
```

**Comment by tucksaun** [comment]
@bob983 your PHP SDK clearly assumes that _embedded should not be provided when there are no results: https://github.com/helpscout/helpscout-api-php/blob/67b91dc41f139882207f2ab034c8caf0928f553f/tests/Http/Hal/Entity/StubPayloads.php#L50-L53

**Comment by bob983** [comment]
@tucksaun thank you for the raw response and also for pointing out that what I feared is most likely correct - the SDK doesn't handle the new empty response well.

Since I'm just a Java dev - I will need to drum up my PHP colleagues. I'll be in touch.

**Comment by ShayCichocki** [comment]
Hey @matt-allan @tucksaun This has been resolved in release 2.3.7 by merging https://github.com/helpscout/helpscout-api-php/pull/176 ! Thanks for the PR
